libsolidity/parsing/Token.h

```diff
@@ -303,10 +305,12 @@ class Token
 	static std::tuple<Token::Value, unsigned short, unsigned short> fromIdentifierOrKeyword(std::string const& _literal);
 
 private:
-	// extractM provides a safe way to extract numbers, 
+	// extractUnsigned provides a safe way to extract numbers, 
 	// if out_of_range error is thrown, they returns 0s, therefore securing 
-	// the variable's identity as an identifier.
-	static unsigned extractM(std::string const& _literal);
+	// the variable's identity as an identifier. If an invalid conversion
+ 	// error is thrown (usually in the case of grabbing N from a fixed type)
+ 	// then a 0 is thrown to purposely ensure that it will declare itself as an identifier
+	static unsigned extractUnsigned(std::string const& _literal);
```

Comment:
Can we change that to take two const iterators (_begin and _end) instead of a string? In that case we do not have to copy the string.